src/main/java/org/opensrp/connector/dhis2/DHIS2ImportOrganizationUnits.java

```diff
@@ -0,0 +1,343 @@
+			System.out.println("type is  : " + type);
+			if (type.equals("MultiPolygon")) {
+				geometry.setType(Geometry.GeometryType.MULTI_POLYGON);
+			} else {
+				geometry.setType(Geometry.GeometryType.valueOf(StringUtils.upperCase(type)));
+			}
+
+		}
+
+		l.setGeometry(geometry);
+
+		l.setJurisdiction(Boolean.TRUE);
+		physicalLocationService.addOrUpdate(l);
+		return l;
+	}
+
+	private LocationTag getOrCreateLocationTag(String tagName) {
+		for (LocationTag tag : locationTagService.getAllLocationTags()) {
+			if (tag.getName().equalsIgnoreCase(tagName)) {
+				return tag;
+			}
+		}
+
+		LocationTag ltag = new LocationTag();
+		ltag.setId(0l);
+		ltag.setName(tagName);
+		return locationTagService.addOrUpdateLocationTag(ltag);
+	}
+
+	private void updateOrCreateAppStateTokens(JSONObject pager) {
+
+		AppStateToken dhisLastSyncPageToken;
+		AppStateToken dhisRowsProcessed;
+		AppStateToken pageCountToken;
+		AppStateToken totalLocationsToken;
+		AppStateToken dhisLocationJobStatus;
+
+		Integer pageCount = (Integer) pager.get("pageCount");
+		Integer totalLocations = (Integer) pager.get("total");
+		DHISImportLocationsJobStatus dhisImportLocationsJobStatus =
+				isJobRunning && !isJobFailed ? DHISImportLocationsJobStatus.RUNNING :
+						!isJobRunning && !isJobFailed ?
+								DHISImportLocationsJobStatus.COMPLETED :
+								DHISImportLocationsJobStatus.FAILED;
+
+		dhisLastSyncPageToken = new AppStateToken("DHIS2-LAST-SYNC-PAGE", pager.get("page"), new Date().getTime());
+		if (allAppStateTokens.findByName("DHIS2-LAST-SYNC-PAGE").size() > 0) {
+			allAppStateTokens.update(dhisLastSyncPageToken);
+		} else {
+			allAppStateTokens.add(dhisLastSyncPageToken);
+		}
+		//
+
+		dhisRowsProcessed = new AppStateToken("DHIS2-LOCATION-ROWS-PROCESSED", rowsProcessed, new Date().getTime());
+		if (allAppStateTokens.findByName("DHIS2-LOCATION-ROWS-PROCESSED").size() > 0) {
+			allAppStateTokens.update(dhisRowsProcessed);
+		} else {
+			allAppStateTokens.add(dhisRowsProcessed);
+		}
+
+		pageCountToken = new AppStateToken("DHIS-PAGE-COUNT", pageCount, new Date().getTime());
+		if (allAppStateTokens.findByName("DHIS-PAGE-COUNT").size() > 0) {
+			allAppStateTokens.update(pageCountToken);
+		} else {
+			allAppStateTokens.add(pageCountToken);
+		}
+
+		totalLocationsToken = new AppStateToken("TOTAL-LOCATIONS", totalLocations, new Date().getTime());
+		if (allAppStateTokens.findByName("TOTAL-LOCATIONS").size() > 0) {
+			allAppStateTokens.update(totalLocationsToken);
+		} else {
+			allAppStateTokens.add(totalLocationsToken);
+		}
+
+		dhisLocationJobStatus = new AppStateToken("DHIS-LOCATIONS-JOB-STATUS",
+				dhisImportLocationsJobStatus != null ? dhisImportLocationsJobStatus.name() : null, new Date().getTime());
+		if (allAppStateTokens.findByName("DHIS-LOCATIONS-JOB-STATUS").size() > 0) {
+			allAppStateTokens.update(dhisLocationJobStatus);
+		} else {
+			allAppStateTokens.add(dhisLocationJobStatus);
+		}
```

Comment:
This block is repeated above 4 other times. Create a function and pass params for tokenName

Reply:
Done